scripts/helpers/wallet-precalculation.helper.js

```diff
@@ -0,0 +1,243 @@
+    'xpub6CA16g2qPwukWAWBMdJKU3p2fQEEi831W3WAs2nesuCzPhbrG29aJsoRDSEDT4Ac3smqSk51uuv6oujU3MAAL3d1Nm87q9GDwE3HRGQLjdP',
+    'xpub6BwNT613Vzy7ARVHDEpoX23SMBEZQMJXdqTWYjQKvJZJVDBjEemU38exJEhc6qbFVc4MmarN68gUKHkyZ3NEgXXCbWtoXXGouHpwMEcXJLf',
+    'xpub6DCyPHg4AwXsdiMh7QSTHR7afmNVwZKHBBMFUiy5aCYQNaWp68ceQXYXCGQr5fZyLAe5hiJDdXrq6w3AXzvVmjFX9F7EdM87repxJEhsmjL',
+    'xpub6CgPUcCCJ9pAK7Rj52hwkxTutSRv91Fq74Hx1SjN62eg6Mp3S3YCJFPChPaDjpp9jCbCZHibBgdKnfNdq6hE9umyjyZKUCySBNF7wkoG4uK',
+  ],
+  walletConfig: {
+    pubkeys: [],
+    total: 5,
+    minSignatures: 3,
+  }
+};
+multisigWalletsData.walletConfig.pubkeys = multisigWalletsData.pubkeys;
+
+export class WalletPrecalculationHelper {
+  WALLETS_FILENAME = '';
+
+  walletsDb = [];
+
+  /**
+   * Initializes the helper with a filename to sync the local wallet storage with.
+   * @param walletsFilename
+   */
+  constructor(walletsFilename) {
+    this.WALLETS_FILENAME = walletsFilename || './tmp/wallets.txt';
+  }
+
+  /**
+   * Generates 22 addresses for a wallet 24-word seed.
+   * @param [params]
+   * @param {string} [params.words] Optional wallet seed words. If empty, generates a new wallet
+   * @param {number} [params.addressIntervalStart=0] Optional interval start index
+   * @param {number} [params.addressIntervalEnd=22] Optional interval end index
+   * @param {{minSignatures:number,wordsArray:string[]}} [params.multisig] Optional multisig object
+   * @returns {{addresses: string[], words: string}}
+   */
+  static generateAddressesForWordsSeed(params = {}) {
+    const timeStart = Date.now().valueOf();
+
+    // Generating seed if none was informed
+    let wordsInput = params.words;
+    if (!wordsInput) {
+      wordsInput = wallet.generateWalletWords();
+    }
+
+    // Calculating addresses
+    const addressIntervalStart = params.addressIntervalStart || 0;
+    const addressIntervalEnd = params.addressIntervalEnd || 22;
+    const addressesArray = [];
+    let multisigDebugData = null;
+    if (params.multisig) {
+      // Multisig calculation
+      const pubkeys = params.multisig.wordsArray.map(w => walletUtils.getMultiSigXPubFromWords(w));
+      for (let i = addressIntervalStart; i < addressIntervalEnd; ++i) {
+        const redeemScript = walletUtils.createP2SHRedeemScript(
+          pubkeys,
+          params.multisig.minSignatures,
+          i
+        );
+        const address = Address.payingTo(Script.fromBuffer(redeemScript), config.network);
+        addressesArray.push(address.toString());
+
+        // Informing debug data
+        multisigDebugData = {
+          words: wordsInput,
+          total: pubkeys.length,
+          minSignatures: params.multisig.minSignatures,
+          pubkeys,
+        };
+      }
+    } else {
+      // Common calculation
+      const xpubkey = walletUtils.getXPubKeyFromSeed(wordsInput, {
+        networkName: config.network,
+        accountDerivationIndex: '0\'/0'
```

Comment:
Semantically this is wrong, because the account derivation index shouldn't have a path. You can use it to simplify things but a comment must exist here to explain that. If we decide to change the `getXPubKeyFromSeed` this might not work.

Reply:
Added an explanatory comment on [1358999](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/13589998c31cf44eb314b340aa469e2f94f473dd):
The idea is that we generate the addresses mimicking the behaviour of generating wallets via the `/start` HTTP post.